daemon/api_debug.go

```diff
@@ -307,6 +327,8 @@ func getDebug(c *Command, r *http.Request, user *auth.UserState) Response {
 		return getChangeTimings(st, chgID, ensureTag, startupTag, all == "true")
 	case "seeding":
 		return getSeedingInfo(st)
+	case "disks":
+		return getDisks(st)
```

Comment:
One (probably silly) question: Is this available to normal users and if so, could we somehow disclose information that normal users would not get via this command (i.e. should we restrict it to "root")?

Reply:
udevadm on my Ubuntu Core device is accessible to normal users, so I don't see any extra information we would be leaking here:

```
ubuntu@ubuntu:~$ udevadm info --query property /dev/vda1
DEVPATH=/devices/pci0000:00/0000:00:03.0/virtio1/block/vda/vda1
DEVNAME=/dev/vda1
DEVTYPE=partition
PARTN=1
PARTNAME=BIOS Boot
MAJOR=252
MINOR=1
SUBSYSTEM=block
USEC_INITIALIZED=5024163
ID_PATH=pci-0000:00:03.0
ID_PATH_TAG=pci-0000_00_03_0
ID_PART_TABLE_UUID=734398f2-f5bd-4d9f-b429-5532e772da10
ID_PART_TABLE_TYPE=gpt
ID_PART_ENTRY_SCHEME=gpt
ID_PART_ENTRY_NAME=BIOS\x20Boot
ID_PART_ENTRY_UUID=107e0b6e-6315-4675-8d41-28322a8f6da6
ID_PART_ENTRY_TYPE=21686148-6449-6e6f-744e-656564454649
ID_PART_ENTRY_NUMBER=1
ID_PART_ENTRY_OFFSET=2048
ID_PART_ENTRY_SIZE=2048
ID_PART_ENTRY_DISK=252:0
DEVLINKS=/dev/disk/by-partlabel/BIOS\x20Boot /dev/disk/by-path/virtio-pci-0000:00:03.0-part1 /dev/disk/by-path/pci-0000:00:03.0-part1 /dev/disk/by-partuuid/107e0b6e-6315-4675-8d41-28322a8f6da6
TAGS=:systemd:
ubuntu@ubuntu:~$ id
uid=1000(ubuntu) gid=1000(ubuntu) groups=1000(ubuntu)
```

Reply:
ah actually trying to implement Maciej's (implicit) suggestion about just calling the disks / gadget functions directly from the snap command led me to find that the one thing we cannot do as a normal user is use blockdev to get the sector size:

```
$ ./snap-exe debug disks
error: cannot get on disk volume for device /dev/nvme0n1: blockdev: cannot open /dev/nvme0n1: Permission denied
```

however I still stand behind my point that the sector size of a disk is not a meaningfully useful privilege escalation :-)